app/Http/Controllers/Api/Volumes/SimilarityIndicesController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Biigle\Http\Controllers\Api\Volumes;
+
+use Biigle\Http\Controllers\Api\Controller;
+use Biigle\Volume;
+use Biigle\MediaType;
+
+class SimilarityIndicesController extends Controller
+{
+    /**
+     * Get all similarity indices of a volume.
+     *
+     * @api {get} volumes/:id/similarityIndices Get similarity indices
+     * @apiGroup Volumes
+     * @apiName VolumeSimilarityIndex
+     * @apiPermission projectMember
+     * @apiDescription Returns a map of image IDs to their similarity indices.
+     *
+     * @apiParam {Number} id The volume ID
+     *
+     * @apiSuccessExample {json} Success response:
+     * {
+     *    "123": 0,
+     *    "321": 1
+     * }
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function index($id)
+    {
+        // should not be find or fail
+        $volume = Volume::where('media_type_id', MediaType::imageId())->findOrFail($id);
+        $this->authorize('access', $volume);
+
+        // if ($volume->images()->whereNull('similarityIndex')->exists()) {
+        //abort(404);
+        //}
```

Comment:
That's the way to do it. To avoid magic numbers you can `use Illuminate\Http\Response` and then call `abort(Response::HTTP_NOT_FOUND)` ([example](https://github.com/biigle/core/blob/3145fffe38a6c98f051a9cfb1f197b0e32752a32/app/Http/Controllers/Api/UserRegistrationController.php#L39)).